python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -176,21 +175,23 @@ class ImageDifferenceConfig(pipeBase.PipelineTaskConfig,
                                                   "of the control sample (AL only)")
     doSubtract = pexConfig.Field(dtype=bool, default=True, doc="Compute subtracted exposure?")
     doPreConvolve = pexConfig.Field(dtype=bool, default=False,
-                                    doc="Convolve science image by its PSF before PSF-matching (AL only)."
-                                    " The difference image becomes a likelihood image.")
+                                    doc="Not in use. Superseded by useScoreImageDetection.",
+                                    deprecated="This option superseded by useScoreImageDetection."
+                                    " Will be removed after v22.")
     useScoreImageDetection = pexConfig.Field(
-        dtype=bool, default=False, doc="Calculate and detect sources on the Zogy score image (Zogy only).")
+        dtype=bool, default=False, doc="Calculate the pre-convolved AL likelihood or "
+        "the Zogy score image. Use it for source detection (if doDetection=True).")
     doWriteScoreExp = pexConfig.Field(
-        dtype=bool, default=False, doc="Write score exposure (Zogy only) ?")
+        dtype=bool, default=False, doc="Write AL likelihood or Zogy score exposure?")
     doScaleTemplateVariance = pexConfig.Field(dtype=bool, default=False,
                                               doc="Scale variance of the template before PSF matching")
     doScaleDiffimVariance = pexConfig.Field(dtype=bool, default=True,
                                             doc="Scale variance of the diffim before PSF matching. "
                                                 "You may do either this or template variance scaling, "
                                                 "or neither. (Doing both is a waste of CPU.)")
-    useGaussianForPreConvolution = pexConfig.Field(dtype=bool, default=True,
-                                                   doc="Use a simple gaussian PSF model for pre-convolution "
-                                                       "(else use fit PSF)? Ignored if doPreConvolve false.")
+    useGaussianForPreConvolution = pexConfig.Field(
```

Comment:
One thing to consider is whether it would be a good idea to raise a ValueError around here, similar to what's done below, if an invalid config is passed (e.g. using zogy and/or useScoreImageDetection is False, and this option is set to True). That said, there's nothing wrong with the code (it will just go back to using the exposure PSF and work fine) though it might lead to confusion down the line in that people are expecting it to be one way when it really is not. Granted, you've covered it in the doc string, so it's not like you're not informing people. Up to you, really.